### DISCUSSION TITLE
Unable to pass ref of clicked element to Drawer

### DISCUSSION THREAD

**Discussion by spidertwin2** [chakra-ui/chakra-ui#3374] [Q&A] [answered]
Hello,

I wanted to display the body of clicked blog title in DrawerBody, but only the last element of blog array is rendered each time.  I think the ref is passed incorrectly, but not sure of the best way to resolve this.  

[Link](https://codesandbox.io/s/5jw45?file=/BlogList.js) to sandbox.

Gracias!

**Comment by chr-ge** [comment, marked as answer]
If you're going to map through each blog and create a `Drawer` for each, you need a `useDisclosure` for each `Drawer`. You can create a new component for each `Drawer` or do this:
```jsx
<Box>
  {blogs.map((blog, i) => {
    const { isOpen, onOpen, onClose } = useDisclosure();

    return(
      <>
        <Text m="10px" key={i} onClick={onOpen} cursor="pointer">
          {blog.title}
        </Text>
        <Drawer
          isOpen={isOpen}
          placement="right"
          onClose={onClose}
          ...
        >
          ...
        </Drawer>
      </>
    )
  })}
</Box>
```

**Reply by dodas** [reply]
I wouldn't recommend doing this.
If number of `blogs` will change, so will the count of times you call `useDisclosure` and that is breaking [rules of hooks](https://reactjs.org/docs/hooks-rules.html).

It's better to wrap it into a new component.

**Reply by chr-ge** [reply]
I agree it would be better to create a separate component.

**Reply by spidertwin2** [reply]
This worked brilliantly! Was so worried it wouldn't be possible. Thanks for the speedy replies! :pray: 

[Link](https://codesandbox.io/s/5jw45?file=/BlogList.js) to updated sandbox.
